sysimage.mk

```diff
@@ -147,7 +158,54 @@ $$(build_private_libdir)/sys$1-o.a $$(build_private_libdir)/sys$1-bc.a : $$(buil
 .SECONDARY: $$(build_private_libdir)/sys$1-o.a $(build_private_libdir)/sys$1-bc.a # request Make to keep these files around
 .SECONDARY: $$(build_private_libdir)/sysbase$1-o.a $(build_private_libdir)/sysbase$1-bc.a # request Make to keep these files around
 endef
+
+# Fat sysimage builder - builds on top of normal sysimage with all stdlibs
+# Parameters: $1=suffix (e.g., -debug), $2=image name (release/debug), $3=opt flags (e.g., -O3), $4=executable
+define sysimg_builder_fat
+build_sysbase_$1 := $$(or $$(CROSS_BOOTSTRAP_SYSBASE),$$(build_private_libdir)/sysbase$1.$$(SHLIB_EXT))
+$$(build_private_libdir)/sys$1-o.a : $$(build_sysbase_$1) $$(BUILDROOT)/stdlib/$2.image $$(JULIAHOME)/base/sysimg_stdlibs.jl
+	@$$(call PRINT_JULIA, cd $$(JULIAHOME)/base && \
+	if ! JULIA_BINDIR=$$(call cygpath_w,$$(build_bindir)) \
+		 WINEPATH="$$(call cygpath_w,$$(build_bindir));$$$$WINEPATH" \
+		 JULIA_LOAD_PATH='@stdlib' \
+		 JULIA_PROJECT= \
+		 JULIA_DEPOT_PATH=$$(call cygpath_w,$$(build_prefix)/share/julia) \
+		 JULIA_NUM_THREADS=1 \
+			$$(call spawn, $4) $3 -C "$$(JULIA_CPU_TARGET)" $$(HEAPLIM) --output-o $$(call cygpath_w,$$@).tmp $$(JULIA_SYSIMG_BUILD_FLAGS) \
+			--startup-file=no --warn-overwrite=yes --depwarn=error --sysimage $$(call cygpath_w,$$<) sysimg_stdlibs.jl; then \
```

Comment:
Isn't this missing `generate_precompile.jl`?

Seems that it never runs on this code path

Reply:
Isn't this using `build_sysbase_$1`, which is not the (non-fat) final base sysimage?

Reply:
This is a bit finicky, e.g.

```
Stdlibs total  ────────────────  5.223498 seconds
Collecting and executing precompile statements
└ Collect (Basic: Assertion failed: ((stream->type == UV_TCP || stream->type == UV_NAMED_PIPE || stream->type == UV_TTY) && "uv_write (unix) does not yet support other types of streams"), function uv__check_before_write, file src/unix/stream.c, line 1297.

[4079] signal 6: Abort trap: 6
in expression starting at /Users/kc/julia/contrib/generate_precompile.jl:10
__pthread_kill at /usr/lib/system/libsystem_kernel.dylib (unknown line)
Allocations: 5369068 (Pool: 5368927; Big: 141); GC: 8
/bin/sh: line 1:  4079 Abort trap: 6     
```